cai_bi/models/dataconsumption/financial/timesheet_entry_aggregates/dim_timesheet_entry_aggregates.sql

```diff
@@ -71,297 +75,889 @@ WITH
-                sum(te.amount_employee) as amount_employee,
-                sum(te.amount_usd) as amount_usd
+                0 as hours_billable,
+                
+                -- get aggregates for employee entity
+                currency_code_employee_entity as currency_code_entity,
+                cast(sum(te.bill_amount_employee_entity) as number(38,2)) as amt_bill_entity,
+                cast(sum(te.cost_amount_employee_entity) as number(38,2)) as amt_cost_entity,
+                cast(sum(te.cost_amount_cola_employee_entity) as number(38,2)) as amt_cost_cola_entity,
+                cast(div0(amt_bill_entity, hours) as number(38,2)) as avg_bill_rate_entity,
+                cast(div0(amt_cost_entity, hours) as number(38,2)) as avg_cost_rate_entity,
+                cast(div0(amt_cost_cola_entity, hours) as number(38,2)) as avg_cost_rate_cola_entity,
+                
+                -- get aggregates for USD
+                currency_code_employee_usd as currency_code_usd,
+                cast(sum(te.bill_amount_usd) as number(38,2)) as amt_bill_usd,
+                cast(sum(te.cost_amount_usd) as number(38,2)) as amt_cost_usd,
+                cast(sum(te.cost_amount_cola_usd) as number(38,2)) as amt_cost_cola_usd,
+                cast(div0(amt_bill_usd, hours) as number(38,2)) as avg_bill_rate_usd,
+                cast(div0(amt_cost_usd, hours) as number(38,2)) as avg_cost_rate_usd,
+                cast(div0(amt_cost_cola_usd, hours) as number(38,2)) as avg_cost_rate_cola_usd
             from base_timesheet_entry as te
             where te.task_name in (select phase_code from time_type_phase_codes where time_type = 'disfun')
             group by all
         union (
-                select key_employee, key_entity, employee_id, date_group_id, date_group_type_id, date_group_year, hours,  avg_rate_employee, avg_rate_usd, amount_employee, amount_usd
+                select key_employee, key_entity, employee_id, date_group_id, date_group_type_id, date_group_year, hours, hours_billable, currency_code_entity, amt_bill_entity, amt_cost_entity, 
+                    amt_cost_cola_entity, avg_bill_rate_entity, avg_cost_rate_entity, avg_cost_rate_cola_entity, currency_code_usd, amt_bill_usd, amt_cost_usd, amt_cost_cola_usd, avg_bill_rate_usd, 
+                    avg_cost_rate_usd, avg_cost_rate_cola_usd
                 from blank_data
             )
         ) as d
         group by all
     ),
 
+    -- ************************************************************************************************************************************************
+    -- DISFUN_DATA_PROJECT_ENTITY: get the sum of all disfun data grouped by project entity limiting to only disfun entries
+    disfun_data_project_entity as (
+         select 'DISFUN' as type, 'Project' as entity_grouping, 106 as type_sort, key_employee, key_entity, employee_id, date_group_id, date_group_type_id, date_group_year,
```

Comment:
Each of these columns should all be on a new line so it's consistent. It is hard to read as one line